app.ts

```diff
@@ -0,0 +1,319 @@
+			(previousBtn as HTMLButtonElement).disabled = false;
+
+			let finalRecord = recordNumberTotal - 1;
+
+			if (fromNumber + numberOfRows >= finalRecord) {
+				fromNumber = finalRecord - numberOfRows;
+				(nextBtn as HTMLButtonElement).disabled = true;
+				(previousBtn as HTMLButtonElement).disabled = false;
+			} else if (fromNumber <= 0) {
+				(nextBtn as HTMLButtonElement).disabled = false;
+				(previousBtn as HTMLButtonElement).disabled = true;
+				fromNumber = 0;
+			}
+
+			toNumber = fromNumber + numberOfRows;
+
+			getRecords(fromNumber, toNumber);
+		}
+	}
+}
+
+window.addEventListener("resize", debounce(resizeScreenData, 500));
+
+class pageNavigation {
+	nextBtn: HTMLElement | null;
+	previousBtn: HTMLElement | null;
+	firstPageBtn: HTMLElement | null;
+	lastPageBtn: HTMLElement | null;
+	confirmationBtn: HTMLElement | null;
+
+	constructor() {
+		this.nextBtn = document.getElementById("next-records-btn");
+		this.previousBtn = document.getElementById("previous-records-btn");
+		this.firstPageBtn = document.getElementById("first-page-btn");
+		this.lastPageBtn = document.getElementById("last-page-btn");
+		this.confirmationBtn = document.getElementById("confirmation-btn");
+
+		if (this.confirmationBtn !== null) {
+			this.confirmationBtn.addEventListener("click", recordSelection);
+		}
+
+		if (this.nextBtn !== null && this.previousBtn !== null && this.firstPageBtn !== null && this.lastPageBtn !== null) {
+			let nextPage = () => {
+				let numberOfRows = Math.floor(window.innerHeight / 50);
+				fromNumber = fromNumber + numberOfRows * count;
+				let toNumber = fromNumber + numberOfRows;
+
+				let finalRecord = recordNumberTotal - 1;
+
+				(this.previousBtn as HTMLButtonElement).disabled = false;
+
+				if (toNumber >= finalRecord) {
+					(this.nextBtn as HTMLButtonElement).disabled = true;
+					fromNumber = finalRecord - numberOfRows;
+				}
+
+				getRecords(fromNumber, toNumber);
+				count = 0;
+			};
+			this.nextBtn.addEventListener("click", () => {
+				count++;
+				nextPage = debounce(nextPage, 500);
+				nextPage();
+			});
+
+			let previousPage = () => {
+				let numberOfRows = Math.floor(window.innerHeight / 50);
+				fromNumber = fromNumber - numberOfRows * count;
+				let toNumber = fromNumber + numberOfRows;
+				(this.nextBtn as HTMLButtonElement).disabled = false;
+
+				if (fromNumber <= 0) {
+					(this.previousBtn as HTMLButtonElement).disabled = true;
+					fromNumber = 0;
+				}
+
+				getRecords(fromNumber, toNumber);
+				count = 0;
+			};
+			this.previousBtn.addEventListener("click", () => {
+				count++;
+				previousPage = debounce(previousPage, 500);
+				previousPage();
+			});
+
+			let firstPage = () => {
+				fromNumber = 0;
+				let numberOfRows = Math.floor(window.innerHeight / 50);
+				let toNumber = fromNumber + numberOfRows;
+				(this.nextBtn as HTMLButtonElement).disabled = false;
+				(this.previousBtn as HTMLButtonElement).disabled = true;
```

Comment:
Same comment about the other buttons, you have already checked the `null`, no need to hard cast, just change the type where you defined these buttons.